File: datastruct-algoritem-java/src/main/java/datastruct/List.java
package com.datastruct;

import com.function.Iterable;

/**
 * @author czx
 * @date 2019/12/24
 */
public interface List<E> extends Iterable<E> {
    boolean add(E e);

    boolean add(int index, E e);

    E get(int index);

    boolean isEmpty();

    E remove(int index);

    int size();

    boolean contains(Object obj);

    int indexOf(Object obj);

    void clear();
}

File: datastruxt-algoritem-go/datastruct/list.go
package datastruct

// Create by czx on 2019/12/25

type List interface {
	Append(e interface{})
	Insert(index int, e interface{})
	Empty() bool
	Size() int
	Capacity() int
	Get(index int) interface{}
	Set(index int, e interface{})
	Remove(index int) (bool, interface{})
	Clear()
}
